ffmpeg.c

```diff
@@ -1365,9 +1448,38 @@ int ffmpeg_put_image(struct ffmpeg *ffmpeg, struct image_data *img_data, const s
         }
 
         /* Setup pointers and line widths. */
-        ffmpeg->picture->data[0] = image;
-        ffmpeg->picture->data[1] = image + (ffmpeg->ctx_codec->width * ffmpeg->ctx_codec->height);
-        ffmpeg->picture->data[2] = ffmpeg->picture->data[1] + ((ffmpeg->ctx_codec->width * ffmpeg->ctx_codec->height) / 4);
+        // Some encoders look for the image in ffmpeg->picture->buf
+        if (ffmpeg->picture->buf[0] == NULL)
+            ffmpeg->picture->data[0] = image;
```

Comment:
~On these lines, how confident are we that the buffer will be NULL?  In the section above, we are looking for a particular codec and then allocating buffers whereas down here we are looking for a NULL in the array elements 0 and 1.~

As I've thought through this, I think it may just be cleaner to look for `h264_v4l2m2m` and just assign NV21 to the buf rather than trying to be too generic.  We don't know what the next customization will require.

For the data pointers, we are no longer setting pointers for the cr/cb like the FFmpeg example in this section but are setting them below.

Reply:
Yes, I think you are right that it will be better to look for `h264_v4l2m2m` and convert format to NV21.

> For the data pointers, we are no longer setting pointers for the cr/cb like the FFmpeg example in this section but are setting them below.

I'm not sure which part you are referring to here.

Reply:
You have addressed this with the latest commit.  It was making sure that the `data[1]` and `data[2]` pointers were explicitly set rather than just setting `data[0]`